File: reservation-service/src/main/java/br/com/wellingtoncosta/reservationservice/data/repository/ReservationRepository.java
package br.com.wellingtoncosta.reservationservice.data.repository;

import br.com.wellingtoncosta.reservationservice.data.entity.ReservationEntity;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

/**
 * @author Wellington Costa on 10/12/18
 */
@Repository
public interface ReservationRepository extends CrudRepository<ReservationEntity, Long> {

    @Query("from ReservationEntity where bookId = :bookId and returnedAt is null")
    ReservationEntity findActiveByBookId(
            @Param("bookId") Long bookId
    );

}

File: reservation-service/src/main/kotlin/br/com/wellingtoncosta/reservation/repository/ReservationRepository.kt
package br.com.wellingtoncosta.reservation.repository

import br.com.wellingtoncosta.reservation.model.Reservation
import org.springframework.data.repository.reactive.ReactiveCrudRepository

/**
 * @author Wellington Costa on 25/07/18.
 */
interface ReservationRepository : ReactiveCrudRepository<Reservation, String>